activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -714,6 +731,22 @@ def test_changing_the_association_id_makes_the_inversed_association_target_stale
 
     assert_equal post2, comment.post
   end
+
+  def test_inverse_record_should_be_set_with_custom_association_primary_key
+    author = Author.create!(name: "Josh")
+    author.essays_2_with_belongs_to_inverse_of.create!
+    author.reload
+
+    assert_same author, author.essays_2_with_belongs_to_inverse_of.first.author
```

Comment:
On main:
![Screenshot 2025-01-22 at 9 54 46 pm](https://github.com/user-attachments/assets/ca39d171-e97e-4e97-b984-7b981a7a29cd)

This is the base `has_one` case that fails unlike its counterparts.